File: BubbleSort/bubbleSort.cpp
#include <algorithm>
#include <iostream>
#include <iomanip>
#include <math.h>
#include <time.h>
using namespace std;
void bubbleSort(int vetor[], int tamanho)
{
	int j = 0;
	bool troca = true;
	int limite = tamanho - 1;

	while (troca)
	{
		troca = false;
		for(int i = 0; i < limite; i++)
		{
			if (vetor[i] > vetor[i+1])
			{
				int trab = vetor[i];
				vetor[i] = vetor[i+1];
				vetor[i+1] = trab;
				j = i;
				troca = true;
			}		
		}
		limite = j;
	}
}


int main()
{
	clock_t tInicio, tFim;
	int size = 99999;
	int vetor[size];
	for(int i = 0; i < size; i++)
		vetor[i] = i;
	random_shuffle ( vetor, vetor + size);

	tInicio = clock();
	bubbleSort(vetor, size);
	tFim = clock();

	double tempo = ((double) (tFim - tInicio) / CLOCKS_PER_SEC);
	cout << fixed << setprecision(5) << tempo << endl;
	return 0;
}

File: BubbleSort/src/bubble_sort.rs
use rand::Rng;
use rand::thread_rng;
use std::time::Instant;

fn bubble_sort(vetor: &mut Vec<i32>, tamanho: usize) {
    let mut j = 0;
    let mut troca = true;
    let mut limite = tamanho - 1;

    while troca {
        troca = false;
        for i in 0..limite {
            if vetor[i] > vetor[i + 1] {
                let trab = vetor[i];
                vetor[i] = vetor[i + 1];
                vetor[i + 1] = trab;
                j = i;
                troca = true;
            }
        }

        limite = j;
    }
}

fn main() {
    let size = 99999;
    let mut array: Vec<i32> = (0..size).collect();
    let mut rng = thread_rng();
    rng.shuffle(&mut array);

    let start = Instant::now();
    bubble_sort(&mut array, size as usize);
    let duration = start.elapsed();

    println!("Bubble Sort: {}ms", duration.as_secs() * 1000 + duration.subsec_nanos() as u64 / 1_000_000);
}
